### DISCUSSION TITLE
Gnet v2.2.0

### DISCUSSION THREAD

**Discussion by panjf2000** [panjf2000/gnet#426] [General]
## Changelogs

- allow creation of multiple engines on the same protocol and port (#419)
- opt: byte alignment, optimization from 144 to 136 byte (#369)
- fix, unix socket bind should be only used in passive mode (#373)
- Fix gnet.Conn not compatible with net.Conn on method Close (#348)
- fix the bug of the return value of conn.WriteTo (#344)

## 🧨 Breaking changes

- opt: redefine AsyncCallback, pass in the error message 691e07747d11e3be90fb3b2996deb19448aa1e0e

## 🚀 Features

- opt: implement Close() for gnet.Conn to make it compatible with net.Conn (#348)
- feat: implement a new API to convert and add net.Conn into gnet.Client c296922d171e64bcd0a97b1bf7eeb7adc747dad6
- allow creation of multiple engines on the same protocol and port (#419)
- feat: add multicast UDP listener support (#412)

## 🛩 Enhancements

- opt: reduce the memory allocation when a connection is idle 27667f3caad72fed296f86a2a5f624253ce01a56

## 🐛 Bugfixes

- bug: set the correct number of bytes written by conn.WriteTo() (#344)
- bug: fix the issue of discarding bytes in buffer mistakenly in Conn.Read() 02820252e2e2a74ece265379386441dd1d4df2d9
- bug: bind socket only when it's in passive mode (#373)
- bug: fix the compile error on linux/mips f2e2fa31adc7694bc45b62aec261ccbe6d77819f
- bug: return io.EOF in Read(), comply with io.Reader and io.Writer 3c3c519b8c57e750231db3d191e7257b5a3cf51d
- bug: fix the issue failing to listen on 0.0.0.0 on linux/arm64 c9eeca1ee0a2d4cce1702d6c529f523fd3786e43
- opt: avoid the long hanging when the peer crashes c138ac53b732540dcff6364d12823f5b6a8f10a8
- opt: put the bytes of net.Addr.Zone back to pool when connection is closed 50406b34552ef6fb86795a9ee6c0e54ad71ae322

## 🗃 Misc

- opt: refine the logic of eventfd in poller 38aa2e0a03ca438c91100c16029ee0b7e6859c07
- opt: fieldalignment, optimization from 144 to 104 byte f70489f555c7f627fcd268907f316dd02425da61
- opt: improve the internal packages of math and bytes f1727999f277e995e8a33319a8e772d9b17b75c9

**Full Changelog**: https://github.com/panjf2000/gnet/compare/v2.0.0...v2.1.0

Thanks to all these contributors: @0-haha, @JemmyH, @ccssrryy, @jdamick, @leki75, @panjf2000, @zhongweikang and zhongweikang for making this release possible.

<hr /><em>This discussion was created from the release <a href='https://github.com/panjf2000/gnet/releases/tag/v2.2.0'>Gnet v2.2.0</a>.</em>

**Comment by symsimmy** [comment]
@panjf2000 最新的2.2版本，好像还是不支持windows启动，有具体的支持计划么

**Reply by panjf2000** [reply]
`gnet` v2 现在已支持 Windows：https://github.com/panjf2000/gnet/releases/tag/v2.3.0-rc.1
